### DISCUSSION TITLE
Proposition: Add border param for floating button.

### DISCUSSION THREAD

**Discussion by laktiv** [chrisgurney/obsidian-note-toolbar#174] [Ideas]
Hi, @chrisgurney!
How do you think, is it possible to add an extra param to CSS to add border for a floating button? As a default white theme user sometimes (more often on mobile) were uncomfortable with the barely seen button. Opacity was not so useful, not universal.

Btw, as an option - it could be done with shadow parameter as well. And is a part of my self learning how to deal with Obsidian, I found shadow samples site and added some to CSS to play. A little excessive, but fun :)

Beginners guide to try:
1. Replace `style.css` file at `.obsidian/plugins/note-toolbar` with the attached file. Do back up. :)
2. You will see this menu in the first place 
![image](https://github.com/user-attachments/assets/a489b03c-35b7-4935-8bff-a5e89a218299)
3. Pick a convenient example
![image](https://github.com/user-attachments/assets/9ae602fe-c789-4683-b6ae-c4b92928c520)
![image](https://github.com/user-attachments/assets/a1f3091c-d11f-46e0-b75f-8ff9e21eb3b8)

[styles.zip](https://github.com/user-attachments/files/17547466/styles.zip)

**Comment by chrisgurney** [comment]
@laktiv You can now add your own custom classes, as defined in CSS snippets, to specific toolbars. See [`1.16-beta-08`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.16-beta-08) and the [documentation](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Custom-styling). Let me know what you think!

**Comment by laktiv** [comment]
Hi @chrisgurney,
Still don't have much time, but I've tested a few functions. I saw only one issue when the styles are not applied to the panel - when you want to apply snippet to menu. Example:
The menu elements (my test was - to change svg color)
![image](https://github.com/user-attachments/assets/c5a5796e-ace5-450f-9bbf-9d0e0f16826a)
Are not applied, because the menu items is not in `.callout[data-callout="note-toolbar"]`
![image](https://github.com/user-attachments/assets/9d206bab-99c6-41c9-9ff1-497aa4ac9530). 
Other looks fine as for first implementation.

Some suggestions for future updates:
1. Make control gui same as device appearance control above (at panels menu). E.g. change from text input to editable list with separate elements. Only 2 classes are visible now, and scrolling in input is not ok, especially for mobiles.
2. Maybe someday it will be possible to apply to ID's (not feature request)
3. Useful to have some indication that the panel has custom styles assigned. I understand that in general this is the complex question for future major updates (to have more convenient gui for controlling panels, menus, visibility and styles), but maybe it's possible to add extra indicator (icon?).

As always - thank you for the updates!

PS: Is it possible to make multi row input text field for script functions/calls params?

**Reply by chrisgurney** [reply]
> PS: Is it possible to make multi row input text field for script functions/calls params?

[Done!](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.16-beta-10) Good idea.

**Reply by chrisgurney** [reply]
> [Custom classes for menu items are] not applied, because the menu items is not in `.callout[data-callout="note-toolbar"]`

Good catch! I'll fix that when I'm able.

**Reply by chrisgurney** [reply]
> 2. Maybe someday it will be possible to apply to ID's (not feature request)

If you inspect toolbars and items, each has an `id` (UUID) that you can target. Or were you thinking of something else?

**Reply by chrisgurney** [reply]
> 3. Useful to have some indication that the panel has custom styles assigned...

I'm curious what you might have in mind here, and what an improved UX showing appearance settings might look like.

**Reply by chrisgurney** [reply]
> 1. Make control gui same as device appearance control above (at panels menu).

What I might do for the custom classes field is make it a textarea below the setting label. Let me try that out.

**Comment by chrisgurney** [comment]
@laktiv 

> [Custom classes for menu items are] not applied, because the menu items is not in .callout[data-callout="note-toolbar"]

This is fixed in the [`1.16-beta-13`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.16-beta-13) release.

And per this:

> Make control gui same as device appearance control above (at panels menu).

I've also made the Custom styles field larger.

**Reply by laktiv** [reply]
Thank you.

**Reply by laktiv** [reply]
Since css class couldn't be applied to note's callout, could they have some unified class by default? As example here (in the end)
```html
<div data-callout-metadata="center|button|between|bigger" data-callout="note-toolbar" class="callout note-bar">
```

**Reply by chrisgurney** [reply]
@laktiv For callouts, you can just make up your own style to the list of styles, and then target it with CSS, like this:

https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Custom-styling#apply-styling-to-callouts

...unless you had another use case that can't be accomplished this way?

**Reply by laktiv** [reply]
Nothing hard, but it looks like extra effort.
Now I'm trying to setup code helper in my vault with snippets. Saw good examples (with NTB btw 😁, [one](https://github.com/MasonGuinn/Codepedia) of them). 

![image](https://github.com/user-attachments/assets/d9832c60-51f2-4463-b103-8cd828b38abc)
Upper toolbar from mapping (settings)
Lower - inline DVJS in text callout which generates buttons in auto mode.

Again, it's easy, no specific request.

**Comment by chrisgurney** [comment]
Now that Note Toolbar supports both custom style classes and the border style for floating buttons (which can be adjusted via Style Settings), I'm going to close this thread.
